src/components/PartnersList/PartnersList.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState, useEffect, useMemo } from 'react'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { PaymentMethod, Region, Service, TeamSize, VerifiedPartner } from '../../interfaces/VerifiedPartner'
+import PartnerCard from '../PartnerCard/PartnerCard'
+import { Dropdown, AccordionTitleProps, Accordion, Form, Menu, CheckboxProps, Button } from 'semantic-ui-react'
+
+import styles from './PartnersList.module.css'
+interface Props {
+  partners: VerifiedPartner[]
+}
+
+enum FilterType {
+  Services = 'services',
+  Region = 'region',
+  TeamSize = 'team_size',
+  PaymentMethod = 'payment_methods',
+  Language = 'languages',
+}
+
+type Filters = Record<FilterType, string[]>
+const EMPTY_FILTER = {} as Filters
+
+const dropdownContent = [
+  {
+    key: FilterType.Services,
+    title: <FormattedMessage id={FilterType.Services} />,
+    options: Service,
+  },
+  {
+    key: FilterType.Region,
+    title: <FormattedMessage id={FilterType.Region} />,
+    options: Region,
+  },
+  {
+    key: FilterType.TeamSize,
+    title: <FormattedMessage id={FilterType.TeamSize} />,
+    options: TeamSize,
+  },
+  {
+    key: FilterType.PaymentMethod,
+    title: <FormattedMessage id={FilterType.PaymentMethod} />,
+    options: PaymentMethod,
+  },
+]
+
+function PartnersList({ partners }: Props) {
+  const [currentIdx, setCurrentIdx] = useState(0)
+  const [filters, setFilters] = useState<Filters>(EMPTY_FILTER)
+  const [filteredPartners, setFilteredPartners] = useState(partners)
+  const [checkBoxState, setCheckBoxState] = useState<Record<string, boolean>>({})
+  const intl = useIntl()
+  const filterText = intl.formatMessage({ id: 'filter' })
+
+  const handleTitleClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>, titleProps: AccordionTitleProps) => {
+    e.stopPropagation()
+    const { index } = titleProps
+    const idxNumber = Number(index)
+    if (currentIdx === idxNumber) {
+      setCurrentIdx(-1)
+    } else {
+      setCurrentIdx(idxNumber)
+    }
+  }
+
+  const handleItemClick = (event: React.MouseEvent<HTMLInputElement, MouseEvent>, itemData: CheckboxProps) => {
```

Comment:
What is this item? Is this a filter? I'd move all the filter logic and ui to a new component or rename all these to reference the Accordion or Filter